app/controlplane/pkg/data/workflowcontract.go

```diff
@@ -372,7 +374,7 @@ func contractInOrgQuery(ctx context.Context, q *ent.OrganizationQuery, orgID uui
 
 	if !c.SkipGetReferences {
 		query = query.WithWorkflows(func(q *ent.WorkflowQuery) {
-			q.Where(workflow.DeletedAtIsNil())
+			q.Where(workflow.DeletedAtIsNil()).WithProject().Select(project.FieldID, project.FieldName)
```

Comment:
I was checking the code and I think it's ok. The `SkipGetReferences` flag already controls whether references are fetched, and when they’re not needed, the fetch is skipped. My reasoning is that projects are part of the reference itself and are always required. To optimize it, let's query the fields necessary to build the reference.